### PR TITLE
Allow for parallel camera access

### PR DESCRIPTION
**Pull request by okiwi6**
## Introduced Changes

Makes the camera access non-mutable so that it can be used from multiple cyclers.
This will be required for other cyclers that want to operate on camera images.

Fixes #

## ToDo / Known Issues
* [x] Test on NAO

*If this is a WIP describe which problems are to be fixed.*

## Ideas for Next Iterations (Not This PR)

*If there are some improvements that could be done in a next iteration, describe them here.*

## How to Test
* upload check if everything still works
* check that images in twix are still displayed correctly

### CODE REVIEW

File: crates/hulk_nao/src/camera.rs [hunk 1]
```diff
@@ -82,9 +101,8 @@ impl Camera {
         bail!("too many unsuccessful waiting retries");
     }
 
-    fn reset(&mut self) -> Result<()> {
+    fn reset(&self) -> Result<()> {
         let _lock = self.i2c_head_mutex.lock();
-        self.camera.take();
```
Comment (schmidma) [inline crates/hulk_nao/src/camera.rs:87]: For the duration of the camera reset, the camera is not locked. What happens to other receivers requesting access to the camera? Should this `reset()` lock the camera for the duration of the reset?

File: crates/vision/src/image_receiver.rs [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
-pub struct CreationContext {}
+pub struct CreationContext {
+    hardware_interface: HardwareInterface,
+}
 
 #[context]
 pub struct CycleContext {
     hardware_interface: HardwareInterface,
     camera_position: Parameter<CameraPosition, "image_receiver.$cycler_instance.camera_position">,
+    cycle_time: AdditionalOutput<Duration, "cycle_time">,
```
Comment (schmidma) [inline crates/vision/src/image_receiver.rs:24]: naming... what do you think of `last_cycle_duration`

**Review by schmidma — CHANGES_REQUESTED** [review]
only minor things, looks good in general

File: crates/vision/src/image_receiver.rs (hunk 2 above)
Comment (okiwi6) [reply crates/vision/src/image_receiver.rs]: Done

File: crates/hulk_nao/src/camera.rs (hunk 1 above)
Comment (okiwi6) [reply crates/hulk_nao/src/camera.rs:87]: `reset()` is private

**Review by schmidma — APPROVED** [review]

**Pull request merged**